### PR TITLE
HealthPass | Partha | modified contract to accept cert details 

### PR DESCRIPTION
**Pull request by partha86**

### CODE REVIEW

File: contracts/HealthPass721.sol [hunk 1]
```diff
@@ -11,24 +11,59 @@ contract HealthPass721 is ERC721URIStorage, Ownable {
...
     
 
-    constructor() ERC721("HealthPass", "HPASS") {
-        
+    constructor(string memory _baseURL) ERC721("HealthPass", "HPASS") {
+        baseURL=_baseURL;
+    }
+
+   //Can be extended based on requirement
+   modifier isUserEligibleForCertificate(address user) {
```
Comment (rajagottumukkala) [inline contracts/HealthPass721.sol:33]: This is interesting. My idea when I created this contract is, we will capture each shot.  Each country/event might have different meaning about what do they mean by fully vaccinated and based on the age too.

File: contracts/HealthPass721.sol [hunk 2]
```diff
@@ -11,24 +11,59 @@ contract HealthPass721 is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
 
+ //We can brainstorm about it
+    struct VaccineDetails {
+     string category;
```
Comment (rajagottumukkala) [inline contracts/HealthPass721.sol:16]: Not a review comment, but curious about what category is?

File: contracts/HealthPass721.sol [hunk 3]
```diff
@@ -11,24 +11,59 @@ contract HealthPass721 is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
 
+ //We can brainstorm about it
```
Comment (rajagottumukkala) [inline contracts/HealthPass721.sol:14]: Just a general question, what are the guidelines around gas based on how much data we store on-chain? Is it possible to move this to attributes which are saved on IPFS?

**Review by rajagottumukkala — COMMENTED** [review]
Great work, I just have one comment (others are questions). Let me know what you think

File: contracts/HealthPass721.sol (hunk 1 above)
Comment (partha86) [reply contracts/HealthPass721.sol:33]: I was just extending it for the app, its like our app takes the ownership of unique minting of each certificate, as of now its a simple, but we can extend it, if we want to have restrictions

File: contracts/HealthPass721.sol (hunk 2 above)
Comment (partha86) [reply contracts/HealthPass721.sol:16]: its just an attribute, something like the main dose or booster dose, or more types

File: contracts/HealthPass721.sol (hunk 3 above)
Comment (partha86) [reply contracts/HealthPass721.sol:14]: True, thats a genuine thing, if we want to have vaccine certificate customisations, then structure will expand, or else some other optimisation technique like compression

**Pull request merged**